src/radical/pilot/agent/executing/funcs.py

```diff
@@ -189,12 +189,12 @@ def _spawn(self, launcher, funcs):
 
     # --------------------------------------------------------------------------
     #
-    def work(self, units):
+    def work(self, tasks):
 
-        if not isinstance(units, list):
-            units = [units]
+        if not isinstance(tasks, list):
+            tasks = [tasks]
 
-        self.advance(units, rps.AGENT_EXECUTING, publish=True, push=False)
+        self.advance(tasks, rps.AGENT_EXECUTING, publish=True, push=False)
 
         for unit in units:
             assert(unit['description']['cpu_process_type'] == 'FUNC')
```

Comment:
`for unit in units` -> `for task in tasks`
`unit['description']...` -> `task['description']...`

Reply:
Thanks, fixed now.